From feb548a6cdea4157bb71dea23fda64fd8e3e16bf Mon Sep 17 00:00:00 2001
From: Palmer Dabbelt <palmer@rivosinc.com>
Date: Mon, 10 Oct 2022 19:14:33 -0700
Subject: [PATCH] 90crypt: Look for dm-crypt devices in by-uuid

I don't get a dm-uuid-CRYPT-LUKS device on my system, which is running
the Gentoo distribution kernel (5.15.32-gentoo-dist).  I'm not sure if
that's a Gentoo-specific thing, but I do get a device in by-uuid and
that name seems pretty general.

Signed-off-by: Palmer Dabbelt <palmer@rivosinc.com>
---
 modules.d/90crypt/parse-crypt.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/90crypt/parse-crypt.sh b/modules.d/90crypt/parse-crypt.sh
index 70e96444..67adeb73 100755
--- a/modules.d/90crypt/parse-crypt.sh
+++ b/modules.d/90crypt/parse-crypt.sh
@@ -166,7 +166,7 @@ else
             if [ $is_keysource -eq 0 ]; then
                 uuid=$luksid
                 while [ "$uuid" != "${uuid#*-}" ]; do uuid=${uuid%%-*}${uuid#*-}; done
-                printf -- '[ -e /dev/disk/by-id/dm-uuid-CRYPT-LUKS?-*%s*-* ] || exit 1\n' "$uuid" \
+                printf -- '[ -e /dev/disk/by-uuid/*%s* ] || exit 1\n' "$uuid" \
                     >> "$hookdir/initqueue/finished/90-crypt.sh"
                 {
                     printf -- '[ -e /dev/disk/by-uuid/*%s* ] || ' "$luksid"
-- 
2.35.1

